e2e/disk_setup.go

```diff
@@ -0,0 +1,104 @@
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func diskSetup() func(*testing.T) {
+	return func(t *testing.T) {
+		namespace := TestNamespace
+
+		// get nodes
+		nodeList := &corev1.NodeList{}
+		err := DeployManagerObj.GetCrClient().List(context.TODO(), nodeList, client.HasLabels{labelNodeRoleWorker})
+		if err != nil {
+			t.Fatalf("failed to list nodes: %+v", err)
+		}
+
+		// represents the disk layout to setup on the nodes.
+		var nodeEnv []nodeDisks
+		for _, disks := range nodeList.Items {
+			nodeEnv = []nodeDisks{
+				{
+					disks: []disk{
+						{size: 10},
+						{size: 20},
+					},
+					node: disks,
+				},
+			}
+		}
+
+		/*
+			// represents the disk layout to setup on the nodes.
+			nodeEnv := []nodeDisks{
+				{
+					disks: []disk{
+						{size: 10},
+						{size: 20},
+					},
+					node: nodeList.Items[0],
+				},
+				{
+					disks: []disk{
+						{size: 10},
+						{size: 20},
+					},
+					node: nodeList.Items[1],
+				},
+			}
+		*/
+		matcher := gomega.NewGomegaWithT(t)
+		gomega.SetDefaultEventuallyTimeout(time.Minute * 10)
+		gomega.SetDefaultEventuallyPollingInterval(time.Second * 2)
+
+		t.Log("getting AWS region info from node spec")
+		_, region, _, err := getAWSNodeInfo(nodeList.Items[0])
+		matcher.Expect(err).NotTo(gomega.HaveOccurred(), "getAWSNodeInfo")
+
+		// initialize client
+		t.Log("initialize ec2 creds")
+		ec2Client, err := getEC2Client(region)
+		matcher.Expect(err).NotTo(gomega.HaveOccurred(), "getEC2Client")
+
+		// create and attach volumes
+		t.Log("creating and attaching disks")
+		createAndAttachAWSVolumes(t, ec2Client, namespace, nodeEnv)
+	}
+
+}
+
+func diskRemoval() func(*testing.T) {
+	return func(t *testing.T) {
```

Comment:
does this need to be a function inside another function ? For simplicity, I would prefer to keep it a single function if internal function is not needed here.